src/components/Dashboard/DashboardSection.js

```diff
@@ -0,0 +1,259 @@
+            className={clsx(
+                classes.section,
+                isNewsSection && classes.sectionNews,
+                isEventsSection && classes.sectionEvents
+            )}
+            id={id}
+        >
+            {showDivider && <Divider classes={{ root: classes.dividerRoot }} />}
+
+            <Typography
+                noWrap
+                gutterBottom
+                variant="h5"
+                component="h5"
+                color="primary"
+            >
+                {name}
+            </Typography>
+
+            <div className={classes.sectionItems}>{uncollapsed}</div>
+
+            <Collapse in={expanded}>
+                <div className={classes.sectionItems}>{collapsible}</div>
+            </Collapse>
+
+            {displayShowMore && (
+                <Button
+                    onClick={() => setExpanded(!expanded)}
+                    className={classes.showMoreBtn}
+                >
+                    <Typography variant="button" display="block">
+                        {expanded ? t("showFewer") : t("showMore")}
+                    </Typography>
+                </Button>
+            )}
+        </div>
+    );
+};
+
+class SectionBase {
+    constructor(kind, name, labelName, idBase) {
+        this.kind = kind;
+        this.name = name;
+        this.label = labelName;
+        this.id = fns.makeID(idBase);
+    }
+
+    getComponent({
+        t,
+        cardWidth,
+        cardHeight,
+        data,
+        showDivider,
+        numColumns,
+        limit,
+    }) {
+        const sorted = data[this.kind][this.name].sort((first, second) => {
+            const firstParsed = Date.parse(first.date_added);
+            const secondParsed = Date.parse(second.date_added);
+
+            let retval;
+
+            // The return values are reversed so we get reverse chronological order.
+            if (firstParsed < secondParsed) {
+                retval = 1;
+            } else if (firstParsed > secondParsed) {
+                retval = -1;
+            } else {
+                retval = 0;
+            }
+
+            return retval;
+        });
+        return (
+            <DashboardSection
+                id={this.id}
+                kind={this.kind}
+                key={`${this.kind}-${this.name}`}
+                items={sorted}
+                name={t(this.label)}
+                section={this.name}
+                showDivider={showDivider}
+                cardWidth={cardWidth}
+                cardHeight={cardHeight}
+                numColumns={numColumns}
+                limit={limit}
+            />
+        );
+    }
+}
+
+export class RecentAnalyses extends SectionBase {
+    constructor() {
+        super(
+            constants.KIND_ANALYSES,
+            constants.SECTION_RECENT,
+            "recentAnalyses",
+            ids.SECTION_RECENT_ANALYSES
+        );
+    }
+
+    getComponent(params) {
+        return super.getComponent(params);
+    }
```

Comment:
If this is just passing through to `super.getComponent`, then can it be omitted in this class (and the others doing the same)?